### PR TITLE
(/integrations/webhooks/sync-data) update guide

### PR DESCRIPTION
**Pull request by victoriaxyz**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1655/integrations/webhooks/sync-data

### This PR:

-

### CODE REVIEW

File: docs/integrations/webhooks/sync-data.mdx [hunk 1]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
-              'svix-signature': svix_signature,
-            }) as WebhookEvent
-          } catch (err) {
-            console.error('Error verifying webhook:', err)
-            return res.status(400).json({ Error: err })
-          }
-
-          // Do something with the payload
-          // For this guide, you simply log the payload to the console
+          // Log payload to console
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:174]:
```suggestion
          // For this guide, log payload to console
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 2]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
 
-          // Create a new Svix instance with your secret.
-          const wh = new Webhook(WEBHOOK_SECRET)
-
           let evt
 
           // Attempt to verify the incoming webhook
           // If successful, the payload will be available from 'evt'
-          // If the verification fails, error out and  return error code
+          // If  verification fails, error out and return error code
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:226]:
```suggestion
          // If verification fails, error out and return error code
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 3]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
-            return res.status(400).json({
+            return void res.status(400).json({
               success: false,
               message: err.message,
             })
           }
 
           // Do something with the payload
-          // For this guide, you simply log the payload to the console
+          // Log payload to console
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:242]:
```suggestion
          // For this guide, log payload to console
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 4]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
           } catch (err) {
             console.log('Error verifying webhook:', err.message)
-            return res.status(400).json({
+            return void res.status(400).json({
               success: false,
               message: err.message,
             })
           }
 
           // Do something with the payload
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:241]:
```suggestion
          // Do something with payload
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 5]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
             }) as WebhookEvent
           } catch (err) {
             console.error('Error verifying webhook:', err)
-            return new Response('Error occured', {
+            return new Response('Verification error', {
               status: 400,
             })
           }
 
           // Do something with the payload
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:173]:
```suggestion
          // Do something with payload
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 6]
```diff
@@ -321,22 +255,22 @@ This guide can be adapted to listen for any Clerk event.
...
   </Tabs>
 
-  ### Narrow the webhook event to get typing
+  ### Narrow to a webhook event for type inference
 
-  The `WebhookEvent` reflects all possible webhook types. You can narrow down the event to get the types inferred correctly for the event type you are working with.
+  `WebhookEvent` encompasses all possible webhook types. Narrow down the event type for accurate typing for specific events.
 
-  In the following example, the `if` statement will narrow the type to the `user.created` type and using `evt.data` will give you autocompletion and type safety for that event.
+  In the following example, the `if` statement narrows the type to `user.created`, enabling type-safe access to evt.data with autocompletion.
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:262]: the guide is about listening for the user.updated event

```suggestion
  In the following example, the `if` statement narrows the type to `user.updated`, enabling type-safe access to evt.data with autocompletion.
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 7]
```diff
@@ -321,22 +255,22 @@ This guide can be adapted to listen for any Clerk event.
...
 
   ```ts {{ filename: 'app/api/webhooks/route.ts', del: [1, 2], ins: [[4, 6]] }}
-  console.log(`Webhook with and ID of ${id} and type of ${eventType}`)
-  console.log('Webhook body:', body)
+  console.log(`Received webhook with ID ${id} and event type of ${eventType}`)
+  console.log('Webhook payload:', body)
 
   if (evt.type === 'user.created') {
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:268]:
```suggestion
  if (evt.type === 'user.updated') {
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 8]
```diff
@@ -347,33 +281,34 @@ This guide can be adapted to listen for any Clerk event.
...
 
-  To trigger the `user.updated` event, you can do one of the following:
+  To trigger the `user.updated` event, you can do either one of the following:
 
-  1. You can edit your user in the Dashboard.
-  1. You can use the `<UserProfile />` component in your application to edit your profile.
+  1. Edit your user in the Clerk Dashboard.
+  1. Select the `<UserProfile />` component in your app to edit your profile.
 
   Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-your-webhook).
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:306]:
```suggestion
  Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-the-webhook).
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 9]
```diff
@@ -105,56 +108,55 @@ This guide can be adapted to listen for any Clerk event.
...
+  For this guide, the payload will be logged to the console. In a real app, you'd use the payload to trigger an action. For example, if listening for the `user.updated` event, you might perform a database `update` or `upsert` to update the user's details.
 
-  Your webhook will need to return either an error code, or a success code, like `200` or `201`. If it returns an error code, the webhook will reflect that in the Dashboard log and it will [retry](/docs/integrations/webhooks/overview#retry). When the webhook returns a success code, there will be no retries and the webhook will show a success status in the Dashboard.
+  The webhook must return either an error code or a success code, such as `200` or `201`. An error code will appear in the Clerk Dashboard log and trigger a [retry](/docs/integrations/webhooks/overview#retry). A success code will stop further retries and show a success status in the Clerk Dashboard log.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:117]:
```suggestion
  The webhook must return either an error code such as `405` or `403` or a success code, such as `200` or `201`. An error code will appear in the Clerk Dashboard log and trigger a [retry](/docs/integrations/webhooks/overview#retry). A success code will stop further retries and show a success status in the Clerk Dashboard log.
```

Maybe link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses ?

File: docs/integrations/webhooks/sync-data.mdx [hunk 10]
```diff
@@ -105,56 +108,55 @@ This guide can be adapted to listen for any Clerk event.
...
 
   > [!NOTE]
-  > The following Route Handler is not specific to the `user.updated` event and can be used for any webhook event you choose to listen to.
+  > The following Route Handler can be used for any webhook event you choose to listen to, not just `user.updated`.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:120]:
```suggestion
  > The following Route Handler will listen to all webhook event types. The next section will demonstrate how to listen to a specific event and also get type inference for that event type.
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 11]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
             }) as WebhookEvent
           } catch (err) {
             console.error('Error verifying webhook:', err)
-            return new Response('Error occured', {
+            return new Response('Verification error', {
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:168]:
```suggestion
            return new Response('Error: Verification error', {
```
Just to be consistent.

File: docs/integrations/webhooks/sync-data.mdx [hunk 12]
```diff
@@ -105,56 +108,55 @@ This guide can be adapted to listen for any Clerk event.
...
 
         export async function POST(req: Request) {
-          // You can find this in the Clerk Dashboard -> Webhooks -> choose the endpoint
-          const WEBHOOK_SECRET = process.env.WEBHOOK_SECRET
+          const SIGNING_SECRET = process.env.SIGNING_SECRET
 
-          if (!WEBHOOK_SECRET) {
-            throw new Error('Please add WEBHOOK_SECRET from Clerk Dashboard to .env or .env.local')
+          if (!SIGNING_SECRET) {
+            throw new Error('Please add SIGNING_SECRET from Clerk Dashboard to .env or .env.local')
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:134]:
```suggestion
            throw new Error('Error: Please add SIGNING_SECRET from Clerk Dashboard to .env or .env.local')
```
Just to be consistent with other errors.

File: docs/integrations/webhooks/sync-data.mdx [hunk 13]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
-          // You can find this in the Clerk Dashboard -> Webhooks -> choose the webhook
-          const WEBHOOK_SECRET = process.env.WEBHOOK_SECRET
-          if (!WEBHOOK_SECRET) {
-            throw new Error('You need a WEBHOOK_SECRET in your .env')
+
+        async (req, res) => {
+          const SIGNING_SECRET = process.env.SIGNING_SECRET
+
+          if (!SIGNING_SECRET) {
+            throw new Error('Please add SIGNING_SECRET from Clerk Dashboard to .env')
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:199]:
```suggestion
            throw new Error('Error: Please add SIGNING_SECRET from Clerk Dashboard to .env')
```

Just to be consistent with other errors.

File: docs/integrations/webhooks/sync-data.mdx [hunk 14]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
...
             evt = wh.verify(payload, {
-              'svix-id': svix_id,
-              'svix-timestamp': svix_timestamp,
-              'svix-signature': svix_signature,
+              'svix-id': svix_id as string,
+              'svix-timestamp': svix_timestamp as string,
+              'svix-signature': svix_signature as string,
             })
           } catch (err) {
             console.log('Error verifying webhook:', err.message)
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:234]:
```suggestion
            console.log('Error: Could not verify webhook:', err.message)
```
Just to be consistent with other errors.

File: docs/integrations/webhooks/sync-data.mdx [hunk 15]
```diff
@@ -163,155 +165,87 @@ This guide can be adapted to listen for any Clerk event.
             }) as WebhookEvent
           } catch (err) {
             console.error('Error verifying webhook:', err)
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:167]:
```suggestion
            console.error('Error: Could not verify webhook:', err)
```
Just to be consistent with other errors.

File: docs/integrations/webhooks/sync-data.mdx [hunk 16]
```diff
@@ -321,22 +255,22 @@ This guide can be adapted to listen for any Clerk event.
...
   </Tabs>
 
-  ### Narrow the webhook event to get typing
+  ### Narrow to a webhook event for type inference
 
-  The `WebhookEvent` reflects all possible webhook types. You can narrow down the event to get the types inferred correctly for the event type you are working with.
+  `WebhookEvent` encompasses all possible webhook types. Narrow down the event type for accurate typing for specific events.
 
-  In the following example, the `if` statement will narrow the type to the `user.created` type and using `evt.data` will give you autocompletion and type safety for that event.
+  In the following example, the `if` statement narrows the type to `user.updated`, enabling type-safe access to evt.data with autocompletion.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:262]: Possibly change this depending on the note above about the `user.*` events.

File: docs/integrations/webhooks/sync-data.mdx [hunk 17]
```diff
@@ -347,33 +281,34 @@ This guide can be adapted to listen for any Clerk event.
...
+  To trigger the `user.updated` event, you can do either one of the following:
 
-  1. You can edit your user in the Dashboard.
-  1. You can use the `<UserProfile />` component in your application to edit your profile.
+  1. Edit your user in the Clerk Dashboard.
+  1. Select the `<UserProfile />` component in your app to edit your profile.
 
-  Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-your-webhook).
+  Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-the-webhook).
+</Steps>
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:307]: Another callout to potentially update this depending on the above

File: docs/integrations/webhooks/sync-data.mdx [hunk 18]
```diff
@@ -347,33 +281,34 @@ This guide can be adapted to listen for any Clerk event.
...
-</Steps>
+1. When you're ready to deploy your app to production, follow [the guide on deploying your Clerk app to production](/docs/deployments/overview).
+1. Create your production webhook by following the steps in the previous [Set up a webhook endpoint](#set-up-a-webhook-endpoint) section. In the **Endpoint URL** field, instead of pasting the ngrok URL, paste your production app URL.
+1. After you've set up your webhook endpoint, you'll be redirected to your endpoint's settings page. Copy the **Signing Secret** and update your `.env.local` file with the new **Signing Secret**.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:313]:
```suggestion
1. After you've set up your webhook endpoint, you'll be redirected to your endpoint's settings page. Copy the **Signing Secret**.
```

Don't update `.env.local` with the production Signing Secret. That would only go in the hosting platform. `.env.local` would retain the initial/development Signing Secret.

**Review by royanger — CHANGES_REQUESTED** [review]
This is looking pretty good. I left some comments and suggestions.

File: docs/integrations/webhooks/sync-data.mdx [hunk 19]
```diff
@@ -1,77 +1,86 @@
...
 
-In this guide, you will learn how to create a Clerk webhook in your Next.js application. You will listen for the `user.updated` event by creating a webhook endpoint in the Clerk Dashboard, creating a webhook handler in your Next.js application, and testing the webhook locally using ngrok and the Clerk Dashboard.
+Clerk offers many user events, but three key events include:
 
-This guide can be adapted to listen for any Clerk event.
+- `user.created`: Triggers when a new user registers in the app or is created via the Clerk Dashboard or Backend API. Listening to this event allows the initial insertion of user information in your database.
+- `user.updated`: Triggers when user information is updated via Clerk components, the Clerk Dashboard, or Backend API. Listening to this event keeps data synced between Clerk and your external database. It is recommended to only sync what you need to simplify this process.
+- `user.deleted`: Triggers when a user deletes their account, or their account is removed via the Clerk Dashboard or Backend API. Listening to this event allows you to delete the user from your database or add a `deleted: true` flag.
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:37]: what was the reasoning behind adding these three and context to them?

Comment (royanger) [reply docs/integrations/webhooks/sync-data.mdx:37]: Prior to review, this guide used just `user.updated` which is a weird spot for a customer to start with. Most customers would want to start with `user.created`. Its not required, but I think mentioning them is valuable. This was the original review comment -> https://github.com/clerk/clerk-docs/pull/1655#discussion_r1823433618

File: docs/integrations/webhooks/sync-data.mdx [hunk 20]
```diff
@@ -1,77 +1,86 @@
...
+  - Configure your production instance
 </TutorialHero>
 
-The recommended way to sync data between Clerk and your application is to use webhooks.
+The recommended way to sync Clerk data to your app is through webhooks.
+
+In this guide, you'll set up a webhook in your Next.js app to listen for the `user.created` event, create an endpoint in the Clerk Dashboard, build a handler, and test it locally using ngrok and the Clerk Dashboard.
 
-In this guide, you will learn how to create a Clerk webhook in your Next.js application. You will listen for the `user.updated` event by creating a webhook endpoint in the Clerk Dashboard, creating a webhook handler in your Next.js application, and testing the webhook locally using ngrok and the Clerk Dashboard.
+Clerk offers many user events, but three key events include:
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:33]:
```suggestion
Clerk offers many events, but three key events include:
```
`user.created`, `user.updated` and `user.deleted` are the only user events. The others are for things like Organizations, Organization Memberships, Organization Invitations, Sessions, etc

File: docs/integrations/webhooks/sync-data.mdx [hunk 21]
```diff
@@ -105,213 +114,144 @@ This guide can be adapted to listen for any Clerk event.
...
+  For this guide, the payload will be logged to the console. In a real app, you'd use the payload to trigger an action. For example, if listening for the `user.created` event, you might perform a database `update` or `upsert` to update the user's details.
 
-  Your webhook will need to return either an error code, or a success code, like `200` or `201`. If it returns an error code, the webhook will reflect that in the Dashboard log and it will [retry](/docs/integrations/webhooks/overview#retry). When the webhook returns a success code, there will be no retries and the webhook will show a success status in the Dashboard.
+  The webhook must return either an [error code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) such as `405` or `403` or a [success code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses), such as `200` or `201`. An error code will appear in the Clerk Dashboard log and trigger a [retry](/docs/integrations/webhooks/overview#retry). A success code will stop further retries and show a success status in the Clerk Dashboard log.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:123]: This should probably be more like:

If the route handler returns a 2xx code that will mark the even as successful and stop retries.
If the route handler returns a 4xx or 5xx code or no code at all, then the webhook event will be retried.

Basically a 2xx/success code stops retries and anything else does not and is considered a failure.

**Review by royanger — CHANGES_REQUESTED** [review]

File: docs/integrations/webhooks/sync-data.mdx [hunk 22]
```diff
@@ -78,27 +76,13 @@ These steps apply to any Clerk event. To make the setup process easier, it's rec
...
-  export const config = {
-    matcher: [
-      // Skip Next.js internals and all static files, unless found in search params
-      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
-      // Always run for API routes
-      '/(api|trpc)(.*)',
-    ],
-  }
-  ```
+  To configure your Middleware, refer to the [clerkMiddleware() guide](docs/references/nextjs/clerk-middleware).
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:81]: looks like you repeat information here. 
also the sentence ends wtih "so the route must be public to allow access". and then the next sentence goes into configuring your middleware.

there should be some relational information that ties these two concepts together:
"so the route must be public to allow access. If you are using `clerkMiddleware()`, you will need to ensure that the `/api/webhooks(.*)` route is public. For more information on configuring routes with `clerkMiddleware()`, see [the reference](/docs/references/nextjs/clerk-middleware).

also don't forget that clerkMiddleware() always needs to be in code snippets. even if it's in a link

File: docs/integrations/webhooks/sync-data.mdx [hunk 23]
```diff
@@ -78,27 +76,13 @@ These steps apply to any Clerk event. To make the setup process easier, it's rec
...
 
   ### Install `svix`
 
-  You'll use [`svix`](https://www.npmjs.com/package/svix) to verify the webhook signature. Run the following command in your terminal to install it:
+  Clerk uses [`svix`](https://www.npmjs.com/package/svix) to deliver webhooks, so you'll need the package to verify the webhook signature. Run the following command in your terminal to install it:
 
   <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:87]: deprecated property - let's do our best to remove it whenever we see it

```suggestion
  <CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 24]
```diff
@@ -116,11 +100,13 @@ These steps apply to any Clerk event. To make the setup process easier, it's rec
...
 
-  The webhook must return either an [error code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) such as `405` or `403` or a [success code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses), such as `200` or `201`. An error code will appear in the Clerk Dashboard log and trigger a [retry](/docs/integrations/webhooks/overview#retry). A success code will stop further retries and show a success status in the Clerk Dashboard log.
+  If the route handler returns a [2xx code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses), the event will be marked as successful, and retries will stop. If the route handler returns a [4xx](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) or [5xx code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses), or no code at all, the webhook event will be [retried](/docs/integrations/webhooks/overview#retry).
+
+  In summary, a 2xx (success) code stops retries, while any other response is considered a failure and will trigger retries.
```
Comment (alexisintech) [inline docs/integrations/webhooks/sync-data.mdx:109]: I don't think there's a need for a "summary" sentence.

```suggestion
  If the route handler returns a [4xx](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) or [5xx code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses), or no code at all, the webhook event will be [retried](/docs/integrations/webhooks/overview#retry). If the route handler returns a [2xx code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses), the event will be marked as successful, and retries will stop.
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 25]
```diff
@@ -1,97 +1,90 @@
...
-  The following example shows the recommended Middleware configuration for your webhook routes.
+  ### Set the webhook route as public in your Middleware
 
-  ```tsx {{ filename: 'middleware.tsx' }}
-  import { clerkMiddleware } from '@clerk/nextjs/server'
+  Incoming webhook events have no auth information. They come from an external source and will not be signed in or signed out. To allow access, the route must be public.
 
-  // Make sure that the `/api/webhooks(.*)` route is not protected here
-  export default clerkMiddleware()
+  Incoming webhook events contain no authentication information. They come from an external source and won't be signed in or signed out, so the route must be public to allow access.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:79]: These are repeated. I believe Alexis left a longer note about this as well.

File: docs/integrations/webhooks/sync-data.mdx [hunk 26]
```diff
@@ -347,33 +271,35 @@ This guide can be adapted to listen for any Clerk event.
...
-  To trigger the `user.updated` event, you can do one of the following:
+  To trigger the `user.created` event, you can do either one of the following:
 
-  1. You can edit your user in the Dashboard.
-  1. You can use the `<UserProfile />` component in your application to edit your profile.
+  1. Edit your user in the Clerk Dashboard.
+  1. Select the `<UserProfile />` component in your app to edit your profile.
 
-  Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-your-webhook).
+  Once you have updated a user, you should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-the-webhook).
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:296]: Maybe not worth changing, but technically this guide doesn't actually update the user. Maybe just this?
```suggestion
  You should be able to see the webhook's payload logged to your terminal. You can also check the Clerk Dashboard to see the webhook attempt, the same way you did when [testing the webhook](#test-the-webhook).
```

File: docs/integrations/webhooks/sync-data.mdx [hunk 27]
```diff
@@ -347,33 +271,35 @@ This guide can be adapted to listen for any Clerk event.
...
-</Steps>
+1. When you're ready to deploy your app to production, follow [the guide on deploying your Clerk app to production](/docs/deployments/overview).
+1. Create your production webhook by following the steps in the previous [Set up a webhook endpoint](#set-up-a-webhook-endpoint) section. In the **Endpoint URL** field, instead of pasting the ngrok URL, paste your production app URL.
+1. After you've set up your webhook endpoint, you'll be redirected to your endpoint's settings page. Copy the **Signing Secret**.
+1. On your hosting platform, update your environment variables by adding **Signing Secret** with the key of `SIGNING_SECRET`.
```
Comment (royanger) [inline docs/integrations/webhooks/sync-data.mdx:304]:
```suggestion
1. On your hosting platform, update your environment variables on your hosting platform by adding **Signing Secret** with the key of `SIGNING_SECRET`.
```

**Review by royanger — APPROVED** [review]
Looks good! I left a couple of super minor notes, but those are easy to fix and then this should be good to ship. 🚀

**Review by alexisintech — APPROVED** [review]

**Pull request merged**